src/scale.js

```diff
@@ -0,0 +1,89 @@
+/* global  WebAnimationsPolyfillExtension */
+
+(function () {
+  function isNumeric (number) {
+    return !isNaN(number);
+  }
+
+  function InvalidArgument (message) {
+    this.message = message;
+    this.name = 'InvalidArgument';
+  }
+
+  function addDefaultAxis (values) {
+    if (values.length === 1) { // x-axis was specified
+      values.push(1, 1);
+    } else if (values.length === 2) { // x and y-axis were specified
+      values.push(1);
+    }
+    return values;
+  }
+
+  function parse (input) {
+    /* According to spec:
+      https://drafts.csswg.org/css-transforms-2/#propdef-scale
+      unspecified scales default to 1
+    */
+    if (input === undefined) {
+      return null;
+    } else if (input === 'none') {
+      return [1, 1, 1];
+    }
+
+    var values = input.split(/\s+/);
+    var numValues = values.length;
+    if (numValues < 1 || numValues > 3) {
+      throw new InvalidArgument('Incorrect number of values for scale');
+    }
+
+    for (var i = 0; i < numValues; i++) {
+      if (!isNumeric(values[i])) {
+        throw new InvalidArgument('Argument must be a number');
```

Comment:
We don't want element.animate() to throw errors, the extension API expects parse to return undefined on failure.